### PR TITLE
Model update: Update the parameter initialization of FC layer of SE-ResNeXt and add a README

### LINKED ISSUE #826: Update the parameter initialization of FC layer of SE-ResNeXt and add a README
The default parameter initialization method will influence  top-1 accuracy in ImageNet2012 classification by 1%. I resolve the problem by updating to new initialization method.

### PR DESCRIPTION
**Pull request by BigFishMaster**
fix #826

### CODE REVIEW

File: fluid/image_classification/README.md [hunk 1]
```diff
@@ -6,3 +6,68 @@ The minimum PaddlePaddle version needed for the code sample in this directory is
...
+1. Download ImageNet-2012 dataset
+```
+# get training set
+wget XXX
+# get validation set
+wget YYY
+# prepare directory
+mkdir ILSVRC2012/
+tar zxf XXX
+tar zxf YYY
```
Comment (qingqing01) [inline fluid/image_classification/README.md]: Better to give the download URL.

File: fluid/image_classification/README.md [hunk 2]
```diff
@@ -6,3 +6,68 @@ The minimum PaddlePaddle version needed for the code sample in this directory is
...
+ILSVRC2012_val_00000006.JPEG 57
+...
+```
+```
+n01440764/n01440764_13575.JPEG 0
+n01440764/n01440764_13581.JPEG 0
+n01440764/n01440764_13602.JPEG 0
+n01440764/n01440764_13625.JPEG 0
+...
+```
```
Comment (qingqing01) [inline fluid/image_classification/README.md]: Can these file lists download in somewhere ? like https://github.com/BVLC/caffe/blob/master/data/ilsvrc12/get_ilsvrc_aux.sh

File: fluid/image_classification/README.md [hunk 3]
```diff
@@ -6,3 +6,68 @@ The minimum PaddlePaddle version needed for the code sample in this directory is
...
+Each line includes an image file path and its label separated by ```SPACE```
+
+## Training a model
+
+To start a training task, one can use command line as:
+
+```
+python train.py --num_layers=50 --batch_size=256 --with_mem_opt=True --parallel_exe=True
+```
```
Comment (qingqing01) [inline fluid/image_classification/README.md]: It's better to give the convergent curve.

File: fluid/image_classification/README.md [hunk 4]
```diff
@@ -6,3 +6,68 @@ The minimum PaddlePaddle version needed for the code sample in this directory is
...
+
+- Top-1/Top-5 Validation Accuracy on ImageNet 2012
+
+The result is obtained by starting with learning rate ```0.1``` and decaying it by ```0.1``` after each ```10``` epoches. The accuracy in table is calculated after ```90``` epoches.
+
+|model | [original paper(Fig.5)](https://arxiv.org/abs/1709.01507) | Pytorch | Paddle fluid
+|- | :-: |:-: | -:
+|SE-ResNeXt-50 | 77.6%/- | 77.71%/93.63% | 77.42%/93.50%
+
+## Finetune a model
```
Comment (qingqing01) [inline fluid/image_classification/README.md]: Need to add the usage here to tell the users how to fine-tune a model.

File: fluid/image_classification/README.md [hunk 5]
```diff
@@ -6,3 +6,68 @@ The minimum PaddlePaddle version needed for the code sample in this directory is
...
+## Training a model
+
+To start a training task, one can use command line as:
+
+```
+python train.py --num_layers=50 --batch_size=256 --with_mem_opt=True --parallel_exe=True
+```
+## Inference
+
+The inference process is conducted after each training epoch.
```
Comment (qingqing01) [inline fluid/image_classification/README.md]: Need a `infer.py` to tell the users how to do inference.

File: fluid/image_classification/se_resnext.py [hunk 6]
```diff
@@ -1,5 +1,36 @@
...
+
+    with init_on_cpu():
+        # update 
+        epoch = ops.floor(global_step / step_each_epoch)
+        decayed_lr = learning_rate * \
+                     (ops.cos(epoch * (math.pi / epochs)) + 1)/2
+    #if global_step % step_each_epoch == 0:
+    #    print("epoch={0}, global_step={1},decayed_lr={2} \
+    #          (step_each_epoch={3})".format( \
+    #          epoch,global_step,decayed_lr,step_each_epoch))
```
Comment (qingqing01) [inline fluid/image_classification/se_resnext.py]: Remove the unused code.

File: fluid/image_classification/se_resnext.py [hunk 7]
```diff
@@ -19,17 +50,28 @@ def conv_bn_layer(input, num_filters, filter_size, stride=1, groups=1,
 def squeeze_excitation(input, num_channels, reduction_ratio):
     pool = fluid.layers.pool2d(
         input=input, pool_size=0, pool_type='avg', global_pooling=True)
+    ### initializer parameter
+    #print >> sys.stderr, "pool shape:", pool.shape
```
Comment (qingqing01) [inline fluid/image_classification/se_resnext.py]: Remove the unused code.

File: fluid/image_classification/se_resnext.py [hunk 8]
```diff
@@ -19,17 +50,28 @@ def conv_bn_layer(input, num_filters, filter_size, stride=1, groups=1,
...
+    stdv = 1.0 / math.sqrt(pool.shape[1] * 1.0)
     squeeze = fluid.layers.fc(input=pool,
                               size=num_channels / reduction_ratio,
-                              act='relu')
+                              act='relu',
+                              param_attr=fluid.param_attr.ParamAttr(
+                                  initializer=fluid.initializer.Uniform(-stdv,
+                                                                        stdv)))
+    #print >> sys.stderr, "squeeze shape:", squeeze.shape
```
Comment (qingqing01) [inline fluid/image_classification/se_resnext.py]: Remove the unused code.

File: fluid/image_classification/se_resnext.py [hunk 9]
```diff
@@ -19,17 +50,28 @@ def conv_bn_layer(input, num_filters, filter_size, stride=1, groups=1,
...
+                                 act='sigmoid',
+                                 param_attr=fluid.param_attr.ParamAttr(
+                                     initializer=fluid.initializer.Uniform(
+                                         -stdv, stdv)))
     scale = fluid.layers.elementwise_mul(x=input, y=excitation, axis=0)
     return scale
 
 
-def shortcut(input, ch_out, stride):
+def shortcut_old(input, ch_out, stride):
```
Comment (qingqing01) [inline fluid/image_classification/se_resnext.py]: If `shortcut_old` is not used, please to remove it.

File: fluid/image_classification/se_resnext.py [hunk 10]
```diff
@@ -112,10 +160,17 @@ def SE_ResNeXt(input, class_dim, infer=False, layers=50):
...
     pool = fluid.layers.pool2d(
-        input=conv, pool_size=0, pool_type='avg', global_pooling=True)
+        input=conv, pool_size=7, pool_type='avg', global_pooling=True)
     if not infer:
-        drop = fluid.layers.dropout(x=pool, dropout_prob=0.2)
+        drop = fluid.layers.dropout(x=pool, dropout_prob=0.5)
     else:
         drop = pool
-    out = fluid.layers.fc(input=drop, size=class_dim, act='softmax')
+    #print >> sys.stderr, "drop shape:", drop.shape
```
Comment (qingqing01) [inline fluid/image_classification/se_resnext.py]: Remove the unused code.

**Review by qingqing01 — COMMENTED** [review]
Need scripts:  infer.py and eval.py.

File: fluid/image_classification/se_resnext.py [hunk 11]
```diff
@@ -1,5 +1,30 @@
...
+import paddle.fluid.layers.control_flow as control_flow
+import paddle.fluid.layers.nn as nn
+import paddle.fluid.layers.ops as ops
+import paddle.fluid.layers.tensor as tensor
+from paddle.fluid.initializer import init_on_cpu
+from paddle.fluid.layers.learning_rate_scheduler import _decay_step_counter
+import math
+
+
+def cosine_decay(learning_rate, step_each_epoch, epochs=120):
```
Comment (qingqing01) [inline fluid/image_classification/se_resnext.py]: 这个函数移动到train.py里吧。

File: fluid/image_classification/train.py [hunk 12]
```diff
@@ -314,12 +327,15 @@ def train_parallel_exe(args,
     # layers: 50, 152
     layers = args.num_layers
     method = train_parallel_exe if args.parallel_exe else train_parallel_do
+    init_model = args.init_model if args.init_model else None
+    pretrained_model = args.pretrained_model if args.pretrained_model else None
```
Comment (qingqing01) [inline fluid/image_classification/train.py:381]: 上面学习率调整测绿没加cosine_decay

**Review by qingqing01 — APPROVED** [review]
先approve了，为了方便cloud验证。 但是文档和代码后续还需要再提升下。
